pkg/connector/roles.go

```diff
@@ -135,16 +136,10 @@ func (o *workspaceRoleType) Grant(ctx context.Context, principal *v2.Resource, e
 		return nil, fmt.Errorf("baton-slack: only users can be assigned a role")
 	}
 
-	// TODO: put the team ID in the entitlement or user ID, not the parent resource
-	if principal.ParentResourceId == nil {
-		l.Warn(
-			"baton-slack: user does not have a parent resource",
-			zap.String("principal_id", principal.Id.Resource),
-		)
-		return nil, fmt.Errorf("baton-slack: user does not have a parent resource")
-	}
+	// teamID is in the entitlement ID at second position
+	teamID := strings.Split(entitlement.Id, ":")[1]
```

Comment:
This is a nit but I'd prefer we do some `len` checking to make sure we have a well-formatted value, and error if that isn't the case.